src/cli/tree.rs

```diff
@@ -0,0 +1,396 @@
+            continue;
+        }
+
+        if let Some(dep) = dep_map.get(dep_name) {
+            let visited = visited_pkgs.contains(&dep.name);
+            visited_pkgs.push(dep.name.to_owned());
+
+            println!(
+                "{}{} {} v{} {}",
+                prefix,
+                symbol,
+                dep.name,
+                dep.version,
+                if visited { "(*)" } else { "" }
+            );
+
+            if visited {
+                continue;
+            }
+
+            let new_prefix = if last {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.empty)
+            } else {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.down)
+            };
+            print_dependency_leaf(dep, new_prefix, dep_map, visited_pkgs);
+        }
+    }
+}
+
+/// Extract the direct Conda and PyPI dependencies from the environment
+fn direct_dependencies(
+    environment: &crate::project::Environment<'_>,
+    platform: &Platform,
+) -> Vec<String> {
+    let mut project_dependency_names = environment
+        .dependencies(None, Some(*platform))
+        .names()
+        .map(|p| p.as_source().to_string())
+        .collect_vec();
+    project_dependency_names.extend(
+        environment
+            .pypi_dependencies(Some(*platform))
+            .into_iter()
+            .map(|(name, _)| name.as_normalized().to_string()),
+    );
+    project_dependency_names
+}
+
+#[derive(Debug)]
+struct Pkg {
+    name: String,
+    version: String,
+    dependencies: Vec<String>,
+}
+
+#[derive(Debug)]
+struct InvertedPkg {
+    name: String,
+    version: String,
+    needed_by: Vec<String>,
+}
+
+/// Builds a hashmap of dependencies, with names, versions, and what they depend on
+fn generate_dependency_map(locked_deps: &Vec<Package>) -> HashMap<String, Pkg> {
+    let mut dep_map = HashMap::new();
+
+    for dep in locked_deps {
+        let version = dep.version().into_owned();
+
+        if let Some(dep) = dep.as_conda() {
+            let name = dep.package_record().name.as_normalized().to_string();
+            let mut dependencies = Vec::new();
+            for d in dep.package_record().depends.iter() {
+                if let Some((dep_name, _)) = d.split_once(' ') {
+                    dependencies.push(dep_name.to_string())
+                }
+            }
+
+            dep_map.insert(
+                name.clone(),
+                Pkg {
+                    name: name.clone(),
+                    version,
+                    dependencies: unique_deps(dependencies),
+                },
+            );
+        } else if let Some(dep) = dep.as_pypi() {
+            let name = dep.data().package.name.to_string();
+
+            let mut dependencies = Vec::new();
+            for p in dep.data().package.requires_dist.iter() {
+                if let Some(markers) = &p.marker {
+                    tracing::info!(
+                        "A bunch of markers on {}, skipping for now {:?}",
```

Comment:
This is not super informative to me, do we need to add markers to the tree command?

Reply:
I haven't been able to figure out which extras or environment markers are active for a package, so I couldn't figure out how to filter it's dependencies to just those that are installed. 

I could include all of the dependencies, and then check if they are in the environment before printing (well before iterating over them), but that might show 'y' was installed by 'x' when really it was an un-specified extra, and instead it was installed by 'z'. Since that's often the reason that I end up using similar commands, I thought it would be probably best to exclude them and have a warning that users can dig into them instead.

Reply:
Ah so there is extra logic needed to find which package actually depends on which extra. Before you could actually add it as a dependency here?

Reply:
Yep, and I'm not sure where to start to figure that out. 

I've tweaked the message to be a bit more clear and to say which package has the dependency with markers.

<img width="1093" alt="image" src="https://github.com/prefix-dev/pixi/assets/1296209/72275d01-bd77-4532-a2d3-2e2b06e001bf">